spark/src/main/scala/geotrellis/spark/io/RDDFilter.scala

```diff
@@ -77,7 +80,10 @@ object Intersects {
         val queryBounds = KeyBounds(
           kb.minKey updateSpatialComponent SpatialKey(bounds.colMin, bounds.rowMin),
           kb.maxKey updateSpatialComponent SpatialKey(bounds.colMax, bounds.rowMax))
-        implicitly[Boundable[K]].intersect(queryBounds, kb).toSeq
+        (queryBounds intersect kb) match {
+          case kb: KeyBounds[K] => List(kb)
+          case EmptyBounds => Nil
+        }
```

Comment:
I was on the fence about this so I left it out. This happens a lot in context of `RDDFilter` shape. So may be a companion helper function here. Do you think its generic enough to be on `Bounds`?